main.py

```diff
@@ -228,15 +230,6 @@ def to_finite(self, mat):
         return final_finite
 
 
-# Utility function adding a value to a queue style iterable, maintaining sorting from smallest to largest
-def min_add(iterable, val):
-    for i, item in enumerate(iterable):
-        if val < item:
-            iterable = iterable[:i] + [val] + iterable[i:(len(iterable) - 1)]
-            return iterable, True
-    return iterable, False
```

Comment:
return type annotions and docs will make returns, especially tuple returns clearer